modules/rpc/yarpc.go

```diff
@@ -36,66 +35,138 @@ import (
 	tch "go.uber.org/yarpc/transport/tchannel"
 )
 
-// YarpcModule is an implementation of a core module using YARPC
-type YarpcModule struct {
+// YARPCModule is an implementation of a core module using YARPC.
+// All the YARPC modules share the same dispatcher and middleware.
+type YARPCModule struct {
 	modules.ModuleBase
-	rpc                      *yarpc.Dispatcher
-	register                 registerServiceFunc
-	config                   yarpcConfig
-	log                      ulog.Log
-	stateMu                  sync.RWMutex
-	inboundMiddlewares       []middleware.UnaryInbound
-	onewayInboundMiddlewares []middleware.OnewayInbound
+	register  registerServiceFunc
+	config    yarpcConfig
+	log       ulog.Log
+	stateMu   sync.RWMutex
+	isRunning bool
 }
 
 var (
-	_dispatcherFn = defaultYarpcDispatcher
+	_dispatcherFn = defaultYARPCDispatcher
 
-	_ service.Module = &YarpcModule{}
+	_ service.Module = &YARPCModule{}
+
+	// Represents the global dispatcher used by all started modules.
+	_dispatcher   *yarpc.Dispatcher
+	_muDispatcher sync.RWMutex
```

Comment:
tiniest of tiny nits: i generally see mutexes named "fooMu" not "muFoo". IOW, consider `_dispatcherMu`